pkg/image/transport.go

```diff
@@ -0,0 +1,266 @@
+/*
+Copyright 2020 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/containers/image/v5/image"
+	"github.com/containers/image/v5/pkg/blobinfocache"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+)
+
+var (
+	cancelTimeout = 300 * time.Second
```

Comment:
Should we write constant names in uppercase to make it clearer?

Reply:
This will make them public. I think they're only relevant in the context of the file. I've seen lowercase constants in other places (e.g. `pkg/importer/registry-datasource.go`).

Reply:
This should be a constant, by the way... :smile:

Reply:
Can you change it to a constant then?

Reply:
Done.